test/exe/Main.hs

```diff
@@ -430,13 +430,13 @@ codeLensesTests = testGroup "code lenses"
 watchedFilesTests :: TestTree
 watchedFilesTests = testGroup "watched files"
   [ testSession "workspace file" $ do
-      _ <- openDoc' "A.hs" "haskell" "module A where"
+      _ <- openDoc' "C/A.hs" "haskell" "module A where"
       RequestMessage{_params = RegistrationParams (List regs)} <- skipManyTill anyMessage (message @RegisterCapabilityRequest)
       let watchedFileRegs =
             [ args | Registration _id WorkspaceDidChangeWatchedFiles args <- regs ]
       liftIO $ assertBool "watches workspace files" $ not $ null watchedFileRegs
   , testSession "non workspace file" $ do
-      _ <- openDoc' "/tmp/A.hs" "haskell" "module A where"
+      _ <- openDoc' "/tmp/A.hs" "haskell" "{-# LANGUAGE NoImplicitPrelude#-}\nmodule A where"
```

Comment:
What changed to require this?

Reply:
Now there is a `DidChangeWatchedFile` subscription message for `Prelude` on various paths, which were not emitted before as they would use relative paths.

Reply:
I’m not following, why do we get a `DidChangeWatchedFile` notification for `Prelude`? Surely we are not editing the contents of that file in this test?

Reply:
`DidChangeWatchedFile` subscriptions are used to implement the `GetFileExists` Shake rule, which is used by `GetLocatedImports` and other Shake rules. So it's not just a matter of whether we are editing the contents of that file in the test.

Reply:
Oh sorry, I mixed up `DidChangeTextDoucment` and `DidChangeWatchedFiles`. Makes sense now :+1:

Reply:
I am hitting this problem in haskell-language-server tests too.  Is the correct solution not to filter out the watch request on non-existent Prelude files?

Reply:
@alanz no, we need to detect if a `Prelude` file starts existing.

Reply:
@cocreature The file monitored as the location of the implicit prelude file is assumed to be in the same directory as the original file. Or in some unlikely location, like deep in the stack workspace, with the project subdirectory appended.

So does it make sense to monitor for that file? Would it not make more sense to monitor wherever the implicit prelude actually exists/comes from?  If that file does suddenly spring into existence, will it replace the implicit prelude?

Reply:
Prelude isn’t all that special in GHC. E.g., you can use `base-noprelude` and provide your own `Prelude` module and GHC will pick that. So I don’t think it makes sense to special case prelude in any way.